File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public static boolean[] visited;
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[n];
        
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                dfs(i, n, computers);
                answer++;
            }
        }
        
        return answer;
    }
    
    public void dfs(int computer, int n, int[][] computers) {
        visited[computer] = true;
        
        for (int i = 0; i < n; i++) {
            if (computers[computer][i] == 1 && !visited[i]) {
                dfs(i, n, computers);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

bool visited[201];

void dfs(int from, int n, vector<vector<int>> &computers) {
    for (int i = 0; i < n; i++) {
        if (from != i && computers[from][i] == 1 && visited[i] == 0) {
            visited[i] = true;
            dfs(i, n, computers);
        }
    }
}

int solution(int n, vector<vector<int>> computers) {
    int cnt = 0;
    for (int i = 0; i < n; i++) {
        if (visited[i]) continue;
        
        cnt++;
        visited[i] = true;
        
        dfs(i, n, computers);
    }
    return cnt;
}